### ISSUE TITLE
Increase number of characters in the reason for appointment field 

### ISSUE THREAD

**Issue by lalexanderson-dsva** [department-of-veterans-affairs/va.gov-team#6711]
## User Story or Problem Statement
As a Veteran who is scheduling or requesting an appointment online, sometimes I need > 100 characters to provide context/reason for my appointment. 

## Goal
We heard from Veterans in user research that sometimes 100 characters is not enough to describe the issue or reason for their appointment. The goal of this user story is to increase (both on the BE and FE of the tool) the number of characters that Veterans can enter when they are requesting an appointment.

## Objectives or Key Results this is meant to further
- Veterans are able to get the right kind of appointment every time they use VA online scheduling

---
## Tasks
- [x] VAMF BE discovery to determine if character limit can be increased for requests and self-scheduled appointments.

- [x] VAMF BE increase character limit to 250
  - https://issues.mobilehealth.va.gov/browse/VAOSR-725
  - https://issues.mobilehealth.va.gov/browse/VAOSR-1407 (var-resources 4.27.0)
- [x] FE update to Request flow (VA & CC)
- [ ] FE update to Direct Schedule flow

## Acceptance Criteria
- [ ] Veterans are able to submit <reason for appointment> that is longer than 100 characters.

**Comment by lenaecb** [comment]
@rtwell @melissajschaff - VAOS has received this feedback for both legacy and now new VAOS. We need to expand the character limit beyond the current 100 character limit. In thinking what that expansion should be, I was wondering if there are some best practices for common text fields such as this. I don't want to allow for a novel to be written. Twitter is 280 and LinkedIn is 700-1200 characters. I want to cognizant of both enough space for the Veteran but also not information overload for the Scheduler. Thoughts?

**Comment by rtwell** [comment]
I am not aware of any best practices regarding character count. I have no problem increasing our current limit (esp because of _why the limit is what it is_) perhaps we meet in the middle of twitter and linkedin at say, 500?

**Comment by melissajschaff** [comment]
I don't know of any hard and fast rules, either, but would be comfortable with 300-500 characters as the limit. 

_**I'm not 100% familiar with the flow, so keep that in mind...**_
I think implementing some helper textm, in addition to a character count **that is displayed and counts down so they can see how many characters are left** to give the Veteran making the request some guidance about what to include for the Scheduler would aid in both reducing info overload and keeping the response concise and straightforward. We could even provide an example of what to write, as helper text. 

**Some ideas/examples**
If the open text field where the Veteran inputs their appointment needs says: _In 1-3 sentences, tell us why you need an appointment._

The helper text could say:
_Idea A:_
>I've had a pain in my right side for 3 days that is constant and is very painful.

_Idea B:_
>Be sure to include: 
>- what part of your body is of concern 
>- when the problem started 
>- have you been treated for this before (etc.)
Basically, list things a scheduler will need to make an informed decision about the appointment that needs to be made, allowing the Veteran to provide enough detail to inform next steps

**Comment by lenaecb** [comment]
Currently new VAOS displays this for the open field. We include some helper text above the box to clarify what should go in the field. We also include a count down below the text field to help Veterans understand how many characters are left. 
![Screen Shot 2020-04-15 at 2 44 05 PM](https://user-images.githubusercontent.com/16602559/79376480-3c946900-7f28-11ea-97f7-132a60f6fc45.png)

I think it may be worth researching what types of messages Scheduler Managers receive. That could advise on if we need to add helper text to direct the Veteran's message more. 

cc: @melissajschaff @rtwell

**Comment by melissajschaff** [comment]
nice -- we're 75% there! 

I agree that we need to know what gets passed to Schedulers in order to make changes. 

@lenaecb does the helper text change based on the option selected above? I wonder if Schedulers need different information based on which type of thing they need to schedule

**Comment by outerpress** [comment]
@laurenernest - caught up on the convo on this one, a couple of notes:
1. The idea of adding helper text based on the questions being asked is interesting. I can add the question about what kind of responses that field generates to the list for things we want to ask schedulers about in the future.

2. This has nothing to do with character count so probably a separate ticket BUT while we're working on this page it might be worth it to remove the second question from the conditional. There's really no value in putting that in a conditional since it shows for every response - it's just a (probably unwelcome) surprise to the user that they need to suddenly answer another question.

3. Back to character count, I am seeing that the character count changes based on the question asked (see video below). This is certainly weird and wrong (that question should just allow the same number of chars for each option) but I tested each option on staging it looks like the count for each is accurate. I'm guessing "more" is the bigger problem here than having a few different characters per option so it's probably fine as is if we can't fix the different count problem.

Character limit changing

https://user-images.githubusercontent.com/2536801/106332545-e042d180-623b-11eb-9b27-893a5d7a07f0.mov

**Comment by laurenernest** [comment]
@outerpress 
1. Great!
2. Agreed, it's weird. Created a shell ticket for it #19360. I assume it has to do with the calculation for the characters remaining after the prefix. @jbalboni can you confirm?
3. The inconsistency is definitely confusing. Related to the reason prefix eating into the total count. There are a few different ways we might be able to handle it and we should explore the feasibility for each. @jbalboni What are the different prefixes we send for each reason/how many characters are used up for each selection?

**Comment by jbalboni** [comment]
For 2, there's no technical reason why the text box has to be conditional on the radio button question. They can be separate. I don't believe there was a specific reason for them being tied together like that, it was just the initial design.

For 3, I want to make sure we're separating out the behavior on the request flow and the direct schedule flow.

For the request flow, the limit is 100 characters. There's nothing special happening behind the scenes for this. The answer to the radio button question goes in one field, the comments in the textbox go into another. The 100 character limit on the textbox is what was recently changed on the backend to 254.

For the direct schedule flow, the only "reason" type field we have to use is something called booking notes. This is a VistA field and it's limited to 150 characters. This limit hasn't changed on the backend. In this case, the limit we show to the user varies depending on the radio button option chosen, because we prefix the booking notes with the answer to the radio button question, and put the comments in the textbox after.

These are the prefix text for the different options:
- Follow-up/Routine (17 characters)
- New issue (9 characters)
- Medication concern (18 characters)
- My reason isn't listed (22 characters)

We also append `: ` to that prefix, so the remaining character count is 150 minus the length of the option selected, minus 2 more characters. So the notes would look something like `New issue: My leg started hurting`. These notes are seen by providers or staff.

**Comment by laurenernest** [comment]
2. Even better! I'll keep the ticket scope for both direct and request to remove the conditional.
3. Thanks for the info!

Attaching the request flow video so we have both.

https://user-images.githubusercontent.com/72046525/106620346-568a5100-6526-11eb-91e0-7b3d9d99bdf5.mov

**Comment by ldelacosta** [comment]
All the issues in this epic has been completed. Closing out the epic. cc: @JRRoof
